filters/auth/authclient.go

```diff
@@ -93,6 +93,9 @@ func (ac *authClient) getTokenintrospect(token string, ctx filters.FilterContext
 
 	if rsp.StatusCode != 200 {
 		io.Copy(io.Discard, rsp.Body)
+		if rsp.StatusCode >= 500 && rsp.StatusCode <= 599 {
```

Comment:
Don't you think that you want to know if status code is >=600? I mean for sure I would like to see if this happens because it's a bug somewhere likely in stdlib.
Maybe also 499 or 0 would be great to check (maybe switch block and return different errors to differentiate them)